core/src/main/java/org/apache/calcite/util/format/postgresql/RomanNumeralMonthFormatPattern.java

```diff
@@ -81,4 +82,49 @@ public RomanNumeralMonthFormatPattern(boolean upperCase, String... patterns) {
       return romanNumeral.toLowerCase(locale);
     }
   }
+
+  @Override protected int parseValue(ParsePosition inputPosition, String input, Locale locale,
+      boolean haveFillMode, boolean enforceLength) throws Exception {
+    final String inputTrimmed = input.substring(inputPosition.getIndex());
+
+    if (inputTrimmed.startsWith(upperCase ? "III" : "iii")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 3);
+      return 3;
+    } else if (inputTrimmed.startsWith(upperCase ? "II" : "ii")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 2);
+      return 2;
+    } else if (inputTrimmed.startsWith(upperCase ? "IV" : "iv")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 2);
+      return 4;
+    } else if (inputTrimmed.startsWith(upperCase ? "IX" : "ix")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 2);
+      return 9;
+    } else if (inputTrimmed.startsWith(upperCase ? "I" : "i")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 1);
+      return 1;
+    } else if (inputTrimmed.startsWith(upperCase ? "VIII" : "viii")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 4);
+      return 8;
+    } else if (inputTrimmed.startsWith(upperCase ? "VII" : "vii")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 3);
+      return 7;
+    } else if (inputTrimmed.startsWith(upperCase ? "VI" : "vi")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 2);
+      return 6;
+    } else if (inputTrimmed.startsWith(upperCase ? "V" : "v")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 1);
+      return 5;
+    } else if (inputTrimmed.startsWith(upperCase ? "XII" : "xii")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 3);
+      return 12;
+    } else if (inputTrimmed.startsWith(upperCase ? "XI" : "xi")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 2);
+      return 11;
+    } else if (inputTrimmed.startsWith(upperCase ? "X" : "x")) {
+      inputPosition.setIndex(inputPosition.getIndex() + 1);
+      return 10;
+    }
+
+    throw new Exception();
```

Comment:
again

Reply:
Changed to a ParseException and added a message.